lib/modules/apostrophe-express/index.js

```diff
@@ -497,40 +497,69 @@ module.exports = {
 
     self.addListenMethod = function() {
       self.apos.listen = function() {
-        // Default address for dev
-        var address = self.options.address || '0.0.0.0';
-        // Default port for dev
-        var port = self.options.forcePort || self.options.port || 3000;
-        // Heroku
-        if (process.env.ADDRESS) {
-          address = process.env.ADDRESS;
-        } else {
-          try {
-            // Stagecoach option
-            address = fs.readFileSync(self.apos.rootDir + '/data/address', 'UTF-8').replace(/\s+$/, '');
-          } catch (err) {
-            console.log("I see no data/address file, defaulting to address " + address);
-          }
+
+        var port;
+        var address;
+
+        if (self.options.forcePort) {
+          port = self.options.forcePort;
+        } else if (self.options.port) {
+          port = self.options.port;
         }
+
         if (!self.options.forcePort) {
           if (process.env.PORT) {
             port = process.env.PORT;
           } else {
-            try {
+            if (fs.existsSync(self.apos.rootDir + '/data/port')) {
```

Comment:
Look at whether throwing an exception here will bubble reasonably or not, are we in a callback?